### ISSUE TITLE
471.omnetpp_ref and 483.xalancbmk_ref failed due to missing system call

### ISSUE THREAD

**Issue by s117** [s117/anycore-riscv#22]
See the log below:

```
spike -m256 -s100000000  +strace=./syscall.trace +std-dump=./console  +chroot=./simenv +target-cwd=/app pk   omnetpp_base.llvm omnetpp.ini
Requesting target memory 0x10000000
******* Resetting core ********** 
****Initializing the processor system****
******* Resetting core ********** 
******* Resetting core ********** 
****Initialization complete****
bad syscall array_size 1063 #98->0!
******* Resetting core ********** 
```

```
$ spike -m512 -s100000000  +strace=./syscall.trace +std-dump=./console  +chroot=./simenv +target-cwd=/app pk   Xalan_base.llvm -v t5.xml xalanc.xsl
Requesting target memory 0x20000000
******* Resetting core ********** 
****Initializing the processor system****
******* Resetting core ********** 
******* Resetting core ********** 
****Initialization complete****
bad syscall array_size 1063 #98->0!
******* Resetting core ********** 
```

**Comment by s117** [comment]
System call 98 is `getrusage`
```
$ ausyscall --dump | grep '^98\s'
98      getrusage
```
I'm not sure whether it makes sense to proxy this call. Looking for a way to disable the use of it in glibc.

**Comment by s117** [comment]
Syscall 98 is `futex` for rv64: https://github.com/s117/riscv-glibc/blob/37694b4bd22eccb79409039b7a38fe1f23a0509d/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h#L64

**Comment by s117** [comment]
`futex` was called because glibc was compiled with multithread support (thread model: posix).

```
$ riscv64-unknown-linux-gnu-clang -v          
clang version 17.0.2 (https://github.com/llvm/llvm-project.git b2417f51dbbd7435eb3aaf203de24de6754da50e)
Target: riscv64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux/bin
Found candidate GCC installation: /home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux/bin/../sysroot/lib/gcc/riscv64-unknown-linux-gnu/13.2.0
Selected GCC installation: /home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux/bin/../sysroot/lib/gcc/riscv64-unknown-linux-gnu/13.2.0

$ riscv64-unknown-linux-gnu-gcc -v  
Using built-in specs.
COLLECT_GCC=riscv64-unknown-linux-gnu-gcc
COLLECT_LTO_WRAPPER=/home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux/libexec/gcc/riscv64-unknown-linux-gnu/13.2.0/lto-wrapper
Target: riscv64-unknown-linux-gnu
Configured with: /home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/riscv-gnu-toolchain/build_linux/../gcc/configure --target=riscv64-unknown-linux-gnu --prefix=/home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux --with-sysroot=/home/s117/Research/toolchain/anycore/gcc13.2.0_clang17.0.2/install-linux/sysroot --with-pkgversion=gc891d8dc23e --with-system-zlib --enable-shared --enable-tls --enable-languages=c,c++,fortran --disable-libmudflap --disable-libssp --disable-libquadmath --disable-libsanitizer --disable-nls --disable-bootstrap --src=../../gcc --disable-multilib --with-abi=lp64d --with-arch=rv64imafd --with-tune=rocket --with-isa-spec=20191213 'CFLAGS_FOR_TARGET=-O2 -g -g  -mcmodel=medlow' 'CXXFLAGS_FOR_TARGET=-O2 -g -g  -mcmodel=medlow'
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 13.2.0 (gc891d8dc23e) 
```

![image](https://github.com/s117/anycore-riscv/assets/8781507/ca7eaf39-8fb3-4a3e-b2a6-8c112b7ef085)

Since PK is a single-threaded environment, multithread support should be disabled in the compiler and glibc.

Related links on how to set the thread model to single:
https://gcc.gnu.org/install/configure.html
https://www.gnu.org/software/libc/manual/html_node/Configuring-and-compiling.html
https://gcc.gnu.org/onlinedocs/gcc-4.8.5/libstdc++/manual/manual/using_concurrency.html
https://llvm.org/docs/CMake.html

**Comment by s117** [comment]
Fixed by [adc2fe2](https://github.com/s117/anycore-riscv/commit/adc2fe2dd874623b515e78d47d6fe82b0bfc0ec6)
